src/linker/Linker/MessageContainer.cs

```diff
@@ -132,6 +132,16 @@ private static MessageContainer CreateWarningMessageContainer (LinkContext conte
 			if (version > context.WarnVersion)
 				return Empty;
 
+			if (origin.MemberDefinition != null) {
+				var assembly = origin.MemberDefinition.DeclaringType.Module.Assembly;
+				var assemblyName = assembly.Name.Name;
+				if (context.AreWarningsCollapsed (assemblyName)) {
+					if (context.WarningAssemblies.Add (assemblyName))
+						context.LogWarning ($"Assembly '{assemblyName}' produced trim warnings", 2104, context.GetAssemblyLocation (assembly));
```

Comment:
Should we set the origin to be the assembly so that the warning can be easily suppressed?
```C#
[assembly: UnconditionalSuppressMessage ("SuppressThis", "IL2104")]
```

Reply:
I'm not sure what should we do here, maybe it's better to leave it as is but I'm curious what the experience should be for a user that's consuming a third-party library that is not yet annotated with `IsTrimmable`, do we expect them to not trim the library at all after seeing this warning, or should we instruct them to turn on granular warnings to decide whether or not they're using code that is dangerous to trim?  And for the later case I suppose that after deciding, the user would get rid of this warning by setting `<NoWarn>IL2104</NoWarn>`?

Reply:
> what the experience should be for a user that's consuming a third-party library that is not yet annotated with `IsTrimmable`
The defaults ("copy") will warning for everything in the assembly, so they'd have a few options:
- stop trimming the app
- trim the assembly and hope the warning goes away (which it might if it originates from unused code)
- turn on granular warnings and investigate

We don't have currently have a good UX around dealing with warnings in assemblies without source code, so they would have to either use NoWarn to suppress the (granular or collapsed) warnings globally, or figure out how to use the attribute XML.

This does set the origin to the assembly path, so you at least get warnings like:
```
/home/sven/.nuget/packages/depnuget/1.0.0/lib/net6.0/depnuget.dll : warning IL2104: Assembly 'depnuget' produced trim warnings
```